checkout-and-start.sh

```diff
@@ -26,5 +26,12 @@ fi
 
 mvn --no-transfer-progress -V -pl war --also-make clean package -DskipTests -Dmaven.test.skip=true -P quick-build
 
+if [[ ${JENKINS_HOME:-} != "" ]]; then
+    export JENKINS_HOME=$JENKINS_HOME
```

Comment:
```suggestion
```

In my local testing this line wasn't necessary. Merely setting `-e JENKINS_HOME=/custom/directory/path` in the `docker run` command will set the environment variable for the entrypoint. And since environment variables in Unix are effectively a mutable per-process map of String to String that is (typically) copied from parent to child when forking, the environment variable is inherited by Java without the need to export it:

```
root@2bfb56c55103:/var/tmp/basil# ps auxwww
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   2504   512 pts/0    Ss   18:28   0:00 /usr/bin/tini -- /checkout-and-start.sh
root           7  0.0  0.0   7112  2052 pts/0    S+   18:28   0:00 /bin/bash /checkout-and-start.sh
root         579 14.5 11.2 14281460 3672248 pts/0 Sl+ 18:29   0:38 java -jar war/target/jenkins.war
root@2bfb56c55103:/var/tmp/basil# cat /proc/1/environ | tr '\0' '\n' | grep JENKINS_HOME
JENKINS_HOME=/var/tmp/basil
root@2bfb56c55103:/var/tmp/basil# cat /proc/7/environ | tr '\0' '\n' | grep JENKINS_HOME
JENKINS_HOME=/var/tmp/basil
root@2bfb56c55103:/var/tmp/basil# cat /proc/589/environ | tr '\0' '\n' | grep JENKINS_HOME
JENKINS_HOME=/var/tmp/basil
```

The only situation where you would need to use `export` is to turn a local variable in your shell into an environment variable in your shell's process, presumably to be inherited by a child you are about to fork.